runner/src/server/services/uploadService.ts

```diff
@@ -239,10 +260,16 @@ export class UploadService {
       };
 
       if (validFiles.length === values.length) {
+        let prefix = applicationId;
+        if (clientSideUploadComponent) {
+          const { id, path } = request.params;
+          prefix = `${prefix}/${id}/${path}/${clientSideUploadComponent.name}`;
+        }
```

Comment:
This is done to maintain backwards compatibility of the existing `FileUploadField` - we only change the prefix given we're using the new component.